src/compositions/useThemeTokens.ts

```diff
@@ -0,0 +1,152 @@
+import { reactive, watch } from 'vue'
+import { useColorTheme } from '@/compositions/useColorTheme'
+
+export function useThemeTokens(): ReturnType<typeof getTokens> {
+  const { value: theme } = useColorTheme()
+
+  const tokens = reactive(getTokens())
+
+  watch(theme, () => Object.assign(tokens, getTokens()))
+
+  return tokens
+}
+
+// this makes more sense to just infer the return type
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+function getTokens() {
+  const style = getComputedStyle(document.documentElement)
```

Comment:
`getComputedStyle` and all these `getPropertyValue` calls might be non-trivial if lots of components were to use the composition. It might make sense to instantiate the reactive var/watcher outside the composition context so repeated calls to the composition share the same reactive variable.